File: 1261.cpp
// 1261. Find Elements in a Contaminated Binary Tree
#include <bits/stdc++.h>
using namespace std;

//  * Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class FindElements {
map<int, bool> node;
public:
    FindElements(TreeNode* root) {
        queue<pair<TreeNode*, int>> que;
        que.push({root, 0});
        while(que.size() > 0){
            TreeNode* curNode = que.front().first;
            int curVal = que.front().second;
            node[curVal] = true;
            que.pop();


            if (curNode -> left) que.push({curNode -> left, 2 * curVal + 1});
            if (curNode -> right) que.push({curNode -> right, 2 * curVal + 2}); 
        }
    }
    
    bool find(int target) {
        if (node[target]) return true;
        return false;
    }
};
    
/**
 * Your FindElements object will be instantiated and called as such:
 * FindElements* obj = new FindElements(root);
 * bool param_1 = obj->find(target);
 */

File: 1261.py
# 1261. Find Elements in a Contaminated Binary Tree
from collections import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class FindElements:

    def __init__(self, root: Optional[TreeNode]):
        self.root = root
        self.nodes = self.recover_tree()

    def recover_tree(self):
        queue = []
        nodes = []
        cur = 0
        queue.append((self.root, cur))
        while(len(queue) > 0):
            cur_node = queue[0][0]
            cur_val = queue[0][1]
            queue.pop(0)
            nodes.append(cur_val)

            if (cur_node.left): queue.append((cur_node.left, 2*cur_val + 1))
            if (cur_node.right): queue.append((cur_node.right, 2*cur_val + 2))

        return nodes
        

    def find(self, target: int) -> bool:
        return target in self.nodes


        


# Your FindElements object will be instantiated and called as such:
# obj = FindElements(root)
# param_1 = obj.find(target)